pkg/sloop/queries/respayloadquery_test.go

```diff
@@ -182,3 +182,33 @@ func Test_removeDupePayloads_twoTheSame_returnsFirstOne(t *testing.T) {
 	ret := removeDupePayloads(input)
 	assert.Equal(t, expected, ret)
 }
+
+func Test_GetResPayload_True_HasSamePrefix(t *testing.T) {
+	untyped.TestHookSetPartitionDuration(time.Hour)
+	partitionId := untyped.GetPartitionId(someTs)
+	values := helper_get_params()
+	expectedKind := "someKind"
+	expectedNS := "someNamespace"
+	expectedName := "someName-11"
+	values[KindParam] = []string{expectedKind}
+	values[NamespaceParam] = []string{expectedNS}
+	values[NameParam] = []string{expectedName}
+
+	var keys []string
+	keys = append(keys, typed.NewWatchTableKey(partitionId, expectedKind, expectedNS, expectedName, someTs).String())
+	keys = append(keys, typed.NewWatchTableKey(partitionId, "someKind", "someNamespace", "someName-1", someTs).String())
+	keys = append(keys, typed.NewWatchTableKey(partitionId, "someKind", "someNamespace", "someName-16", someTs).String())
+	keys = append(keys, typed.NewWatchTableKey(partitionId, "someKind", "someNamespace", "someName-15", someTs).String())
+
+	tables := helper_get_resPayload(keys, t, somePTime)
+	res, err := GetResPayload(values, tables, someTs.Add(-1*time.Hour), someTs.Add(6*time.Hour), someRequestId)
+	assert.Nil(t, err)
+	expectedRes := `[
+ {
+  "payloadTime": 1546398245000000006,
+  "payload": "{\n  \"metadata\": {\n    \"name\": \"someName\",\n    \"namespace\": \"someNamespace\",\n    \"uid\": \"6c2a9795-a282-11e9-ba2f-14187761de09\",\n    \"creationTimestamp\": \"2019-07-09T19:47:45Z\"\n  }\n}",
```

Comment:
nit: shouldn't the name here be someName-1?

Reply:
not a nit - it should match the NameParam inserted in the `values` at line 195

Reply:
So that is because the sample payload used for all test cases is defined [here](https://github.com/salesforce/sloop/blob/master/pkg/sloop/queries/respayloadquery_test.go#L22) . The payload key is what needs to be correct in this case